homeassistant/components/image_processing/opencv.py

```diff
@@ -0,0 +1,120 @@
+"""
+Component that performs OpenCV classification on images.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/image_processing.opencv/
+"""
+from datetime import timedelta
+import logging
+
+from homeassistant.core import split_entity_id
+from homeassistant.components.image_processing import (
+    ImageProcessingEntity,
+    PLATFORM_SCHEMA,
+)
+from homeassistant.components.opencv import (
+    ATTR_MATCHES,
+    CLASSIFIER_GROUP_CONFIG,
+    CONF_CLASSIFIER,
+    CONF_ENTITY_ID,
+    CONF_NAME,
+    process_image,
+)
+
+DEPENDENCIES = ['opencv']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_TIMEOUT = 10
+
+SCAN_INTERVAL = timedelta(seconds=2)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(CLASSIFIER_GROUP_CONFIG)
+
+
+def _create_processor_from_config(hass, camera_entity, config):
+    """Create an OpenCV processor from configurtaion."""
+    classifier_config = config[CONF_CLASSIFIER]
+    name = '{} {}'.format(
+        config[CONF_NAME],
+        split_entity_id(camera_entity)[1].replace('_', ' '))
+
+    processor = OpenCVImageProcessor(
+        hass,
+        camera_entity,
+        name,
+        classifier_config,
+    )
+
+    return processor
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the OpenCV image processing platform."""
+    if discovery_info is None:
+        return
+
+    devices = []
+    for camera_entity in discovery_info[CONF_ENTITY_ID]:
+        devices.append(
+            _create_processor_from_config(
+                hass,
+                camera_entity,
+                discovery_info))
+
+    add_devices(devices)
+
+
+class OpenCVImageProcessor(ImageProcessingEntity):
```

Comment:
Here are classifiers that people can use (without even getting into making their own):
https://github.com/opencv/opencv/tree/master/data/haarcascades
https://github.com/opencv/opencv/tree/master/data/lbpcascades

Examples include:
Cat, Cat Face, Silverware, Upper Body, Lower Body, Full Body, License Plates

I do not agree with making this a `ImageProcessingFaceEntity` as that would be misleading to what it can process.